test-gensalt-extradata.c

```diff
@@ -85,6 +82,7 @@ static const struct testcase testcases[] =
 #if INCLUDE_gost_yescrypt
   { "$gy$", 7, 11, },
 #endif
+  { "", 0, 0, }
```

Comment:
Why is this change necessary?

Reply:
If we configure with `--enable-hashes=bcrypt_x` explicitly, the `testcases` array needs at least one (and in this case the terminating) element.

Reply:
What's wrong with the testcases array having zero elements?

Reply:
Oh, right, ISO C forbids zero-length arrays.  In that case, please change `const char prefix[8];` in `struct testcase` to `const char *prefix;` so that we can use `{ 0, 0, 0 }` for the terminator (avoiding potential confusion with the `""` prefix used for descrypt) and make the main loop condition be `for (size_t i = 0; testcases[i].prefix; i++)` and then `i` can remain a `size_t`.

Reply:
Changed in rebased commit.